presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -664,6 +665,11 @@ public SystemSessionProperties(
                         OPTIMIZED_REPARTITIONING_ENABLED,
                         "Experimental: Use optimized repartitioning",
                         featuresConfig.isOptimizedRepartitioningEnabled(),
+                        false),
+                booleanProperty(
+                        SIMPLIFY_ARRAY_OPERATIONS,
```

Comment:
`SIMPLIFY_ARRAY_OPERATIONS` probably is a little generic? probably `OPTIMIZE_FILTER_CARDINALITY` will be more specific for user?

Reply:
Alternative question to ask: Do you want this flag to guard all future optimization rules that would be added to arrays?

Reply:
> Alternative question to ask: Do you want this flag to guard all future optimization rules that would be added to arrays?

Yeah - I was thinking about that. It's a tricky balance between having too many flags and controlling what is enabled. Since most of these optimizations are opportunistic and hopefully don't need to be turned off frequently, I thought one is just enough. But i can change it if there is a strong reason.